src/wazuh_modules/agent_upgrade/manager/wm_agent_upgrade_commands.c

```diff
@@ -67,6 +67,19 @@ STATIC int wm_agent_upgrade_analyze_agent(int agent_id, wm_agent_task *agent_tas
  * */
 STATIC int wm_agent_upgrade_validate_agent_task(const wm_agent_task *agent_task, const wm_manager_configs* manager_configs) __attribute__((nonnull));
 
+void wm_agent_upgrade_cancel_pending_upgrades() {
+    cJSON *cancel_request = NULL;
+    cJSON *cancel_response = NULL;
+
+    cancel_response = cJSON_CreateArray();
+    cancel_request = wm_agent_upgrade_parse_task_module_request(WM_UPGRADE_CANCEL_TASKS, NULL, NULL, NULL);
+
+    wm_agent_upgrade_task_module_callback(cancel_response, cancel_request, NULL, NULL);
+
```

Comment:
Here we could read the response and throw an `mterror` in case something went wrong on the task module